aggregate_spatial.json

```diff
@@ -1,7 +1,7 @@
 {
     "id": "aggregate_spatial",
     "summary": "Zonal statistics for geometries",
-    "description": "Aggregates statistics for one or more geometries (e.g. zonal statistics for polygons) over the spatial dimensions. This process passes a list of values to the reducer. In contrast, ``aggregate_spatial_binary()`` passes two values, which may be better suited especially for UDFs in case the number of values gets too large to be processed at once.\n\nThe data cube must have been reduced to only contain two spatial dimensions and a third dimension the values are aggregated for, for example the temporal dimension to get a time series. Otherwise, this process fails with the `TooManyDimensions` exception.\n\nThe number of total and valid pixels is returned together with the calculated values.",
+    "description": "Aggregates statistics for one or more geometries (e.g. zonal statistics for polygons) over the spatial dimensions.\n\nThis process passes a list of values to the reducer. The list of values has an undefined order, therefore processes such as ``last()`` and ``first()`` that depend on the order of the values will lead to unpredictable results. In contrast, ``aggregate_spatial_binary()`` passes two values, which may be better suited especially for UDFs in case the number of values gets too large to be processed at once. An 'unbounded' aggregation over the full extent of the horizontal spatial dimensions can be computed with the process ``reduce_spatial()``.\n\nThe data cube must have been reduced to only contain two spatial dimensions and a third dimension the values are aggregated for, for example the temporal dimension to get a time series. Otherwise, this process fails with the `TooManyDimensions` exception.\n\nThe number of total and valid pixels is returned together with the calculated values.",
```

Comment:
I know this is the original text, but there is room for improvement here:
> The data cube must have been reduced to only contain two spatial dimensions and a third dimension the values are aggregated for, for example the temporal dimension to get a time series

With "must have been reduced" it sounds that the user must do something before using `aggregate_spatial`, while in most use cases (all uses cases as far as I've seen) the input data will be just fine as-is.

Also: in the VITO backend we *do* support `aggregate_spatial` on a 4D cube ("t", "bands", "x", "y") and return a "3D" vector cube ("t", "band", "geometry_id").  So I'm not sure this note on `TooManyDimensions` is actually necessary .

Reply:
Most collection I've seen was x,y,t,bands, so users would need to reduce before. So I think the language is fine as it is.

Yeah, we were pretty strict at the beginning by only allowing 3D. I'm not sure all back-ends can/want to support 4D? Anyway, this is the issue to discuss it: #126